pyi.py

```diff
@@ -1592,23 +1594,32 @@ def _check_class_bases(self, bases: list[ast.expr]) -> None:
             self.error(Generic_basenode, Y060)
 
     def visit_ClassDef(self, node: ast.ClassDef) -> None:
+        class_is_protocol = class_is_typeddict = False
+        for base in node.bases:
+            if _is_Protocol(base):
+                class_is_protocol = True
+            if _is_TypedDict(base):
```

Comment:
a base cannot be both a `TypedDict` _and_ a `Protocol`:

```suggestion
            elif _is_TypedDict(base):
```

Reply:
I actually don't like that change. The current code is obviously correct; to verify that the elif version is correct, you need to have the background knowledge that a class can't be both a TypedDict and a Protocol.

Reply:
Fair enough!